source/javascripts/app/_search.js

```diff
@@ -29,22 +29,36 @@
     });
   }
 
+  var wait = (
+      function(){
+        var timeout_handle = 0;
+        return function(executing_function, wait_time){
+          clearTimeout (timeout_handle);
+          timeout_handle = setTimeout(executing_function, wait_time);
+        };
+      })();
+
```

Comment:
bounce and delay have different use cases. I did not found a proper `lib` to place it so left it at _search.js